app/shop.py

```diff
@@ -0,0 +1,57 @@
+from typing import List, Dict, Union
+from datetime import datetime
+from unittest.mock import MagicMock
+
+from app.customer import Customer
+from app.trip import Trip
+
+
+class Shop:
+
+    def __init__(
+        self,
+        name: str,
+        location: List[int],
+        products: Dict[str, Union[int, float]]
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    @staticmethod
+    def convert_to_int_if_possible(number: float) -> float | int:
+        if number == int(number):
+            return int(number)
+        return number
+
+    @staticmethod
+    def shop_receipt(
+        customer: Customer,
+        trip_instances: List[Trip]
+    ) -> None:
+        cheapest_trip = Trip.cheapest_trip(trip_instances)
+        datetime_mock = MagicMock(wrap=datetime)
+        datetime_mock.now.return_value = datetime(2021, 1, 4, 12, 33, 41)
+
+        milk_cost = Shop.convert_to_int_if_possible(
+            cheapest_trip.total_milk_cost
+        )
+        bread_cost = Shop.convert_to_int_if_possible(
+            cheapest_trip.total_bread_cost
+        )
+        butter_cost = Shop.convert_to_int_if_possible(
+            cheapest_trip.total_butter_cost
+        )
```

Comment:
hardcode 

What if some customer need fish or chicken for example?

Reply:
fixed it